### DISCUSSION TITLE
This project is based on assumption the cloud functions and cloud firestore belong to same project

### DISCUSSION THREAD

**Discussion by mSaurabh** [firebase/functions-samples#843] [Q&A] [answered]
## What if I am making a call to a cloud function that is a part of a different project ? How do I authenticate that user ?

*_Ex_*: 
**Project 1** -- all my users are a part of this 

**Project 2** -- specialized only for cloud functions

How can I initialize the second project using auth.admin? 
Can I pass the service account info (project id and service account id) via a fetch call from Project 1 
and use it in Project 2 and initialize the Project 1 app and authenticate my users ? 
or is callable function better ? 

I am happy to add more clarity if my question isn't clear.

**Comment by samtstern** [comment, marked as answer]
@mSaurabh function triggers cannot span projects.  

However you can use `firebase-admin` across projects, as you said you'll just need to provide service account credentials. If you want you can use Cloud IAM to authorize one single service account to access multiple projects.
